DOC-637: apply latest changes from ref manual (#1961)
hazelcast/hz-docs: docs/modules/clients/pages/cplusplus.adoc

=== Install from source code using CMake
==== Requirements
1. Linux, macOS or Windows
2. A compiler that supports C++11 or above: The project detects the minimum C++ standard supported by the Boost library
and uses the same language level for compilation. If you are using Boost 1.82.0 or higher version, the project requires

> C++14 or above since Boost.Math requires this language level as minimum starting 1.82.0 version.

Reviewer: ```suggestion
C++14 or later because Boost.Math requires this language level to be a minimum of 1.82.0.
```

> For lower Boost versions, C++11 or above support is sufficient.

Reviewer: ```suggestion
For lower Boost versions, C++11 or later is sufficient.
```

```sh
g++ -DHZ_BUILD_WITH_SSL -DBOOST_CHRONO_DYN_LINK -DBOOST_CHRONO_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DBOOST_THREAD_VERSION=5 -I/var/git/hazelcast-cpp-client/build/include -std=gnu++11 -c main.cpp
```
===== Build the hazelcast-cpp-client library with vcpkg toolchain

If you want to build the `hazelcast-cpp-client` library with vcpkg toolchain, you can use the following command

> (For windows, please use `cmake.exe` instead of `cmake`):

Reviewer: ```suggestion
(For Windows, use `cmake.exe` instead of `cmake`):
```

```sh
cmake -B build -DCMAKE_TOOLCHAIN_FILE=<path to vcpkg folder>/scripts/buildsystems/vcpkg.cmake
cmake --build build --config Release
```

> if you want to build the project with tests and examples enabled, then you can use the following command:

Reviewer: ```suggestion
If you want to build the project with tests and examples enabled, use the following command:
```
